internal/nursery/reverse.go

```diff
@@ -235,36 +282,64 @@ func (nursery *Nursery) handleReverseSwapStatus(reverseSwap *database.ReverseSwa
 	nursery.sendReverseSwapUpdate(*reverseSwap)
 }
 
-func (nursery *Nursery) handleReverseSwapDirectPayment(swap *database.ReverseSwap, output *onchain.Output) {
-	err := nursery.database.RunTx(func(tx *database.Transaction) error {
-		logger.Debugf("Found direct payment to Reverse Swap %s", swap.Id)
-		if swap.ClaimTransactionId != "" && swap.ClaimTransactionId != output.TxId {
-			return fmt.Errorf("claim transaction changed: %s", swap.ClaimTransactionId)
+func (nursery *Nursery) chooseDirectOutput(swap *database.ReverseSwap, feeEstimations boltz.FeeEstimations, outputs []*onchain.Output) (*onchain.Output, error) {
+	slices.SortFunc(outputs, func(a, b *onchain.Output) int {
+		return cmp.Compare(a.Value, b.Value)
+	})
+
+	for _, output := range outputs {
+		found, err := nursery.database.QueryReverseSwapByClaimTransaction(output.TxId)
+		if err != nil && !errors.Is(err, sql.ErrNoRows) {
+			return nil, fmt.Errorf("could not query reverse swap by claim transaction: %s", err)
+		}
+		if found != nil && found.Id != swap.Id {
+			logger.Infof("Direct Transaction is claim of %s", found.Id)
+			continue
 		}
 
-		currency := swap.Pair.To
-		pairs, err := nursery.boltz.GetReversePairs()
-		if err != nil {
-			return fmt.Errorf("get reverse pairs: %s", err)
+		// we dont know the output value of liquid when the claim address is external
+		if swap.Pair.To == boltz.CurrencyLiquid && swap.WalletId == nil && output.Value == 0 {
```

Comment:
Should we just accept the payment in this case? Seems like an easy exploit
```suggestion
		// We don't know the output value of liquid when the claim address is external
		if swap.Pair.To == boltz.CurrencyLiquid && swap.WalletId == nil && output.Value == 0 {
```